java/client/src/main/java/glide/api/BaseClient.java

```diff
@@ -801,9 +802,21 @@ public CompletableFuture<GlideString[]> mget(@NonNull GlideString[] keys) {
     }
 
     @Override
-    public CompletableFuture<String> mset(@NonNull Map<String, String> keyValueMap) {
-        String[] args = convertMapToKeyValueStringArray(keyValueMap);
-        return commandManager.submitNewCommand(MSet, args, this::handleStringResponse);
+    public <ArgType> CompletableFuture<String> mset(@NonNull Map<ArgType, ArgType> keyValueMap) {
+        if (!keyValueMap.isEmpty()) {
+            var key = keyValueMap.keySet().iterator().next();
+            if (key instanceof String) {
+                String[] args = convertMapToKeyValueStringArray((Map<String, String>) keyValueMap);
```

Comment:
You can reduce the code complexity here:

```java
GlideString[] args = new GlideString[0];
if ((key instanceof String) || (key instanceof GlideString)) {
  args = flattenMapToGlideStringArray(keyValueMap);
} else {
  throw ...
}
return commandManager.submitNewCommand(MSet, new GlideString[0], this::handleStringResponse);
```

Reply:
This doesn't work, I've tried that previously. In normal string tests we get null pointer exception in this version